src/main/java/com/box/sdk/BoxStoragePolicyAssignment.java

```diff
@@ -0,0 +1,273 @@
+        URL url = STORAGE_POLICY_ASSIGNMENT_URL_TEMPLATE.build(api.getBaseURL());
+        BoxJSONRequest request = new BoxJSONRequest(api, url, HttpMethod.POST);
+        JsonObject requestJSON = new JsonObject()
+            .add("storage_policy", new JsonObject()
+                .add("type", "storage_policy")
+                .add("id", policyID))
+            .add("assigned_to", new JsonObject()
+                .add("type", "user")
+                .add("id", userID));
+
+        request.setBody(requestJSON.toString());
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+
+        BoxStoragePolicyAssignment storagePolicyAssignment = new BoxStoragePolicyAssignment(api,
+                responseJSON.get("id").asString());
+
+        return storagePolicyAssignment.new Info(responseJSON);
+    }
+
+    /**
+     * Updates the information about the BoxStoragePolicyAssignment with any info fields that have been
+     * modified locally.
+     * @param info the updated info.
+     */
+    public void updateInfo(BoxStoragePolicyAssignment.Info info) {
+        URL url = STORAGE_POLICY_ASSIGNMENT_WITH_ID_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        BoxJSONRequest request = new BoxJSONRequest(this.getAPI(), url, "PUT");
+        request.setBody(info.getPendingChanges());
+
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+        info.update(responseJSON);
+    }
+
+    /**
+     * Returns a BoxStoragePolicyAssignment information.
+     * @param api                the API connection to be used by the resource.
+     * @param resolvedForType    the assigned entity type for the storage policy.
+     * @param resolvedForID      the assigned entity id for the storage policy.
+     * @return information about this {@link BoxStoragePolicyAssignment}.
+     */
+    public static BoxStoragePolicyAssignment.Info getAssignmentForTarget(final BoxAPIConnection api,
+                                                                         String resolvedForType, String resolvedForID) {
+        QueryStringBuilder builder = new QueryStringBuilder();
+        builder.appendParam("resolved_for_type", resolvedForType)
+               .appendParam("resolved_for_id", resolvedForID);
+        URL url = STORAGE_POLICY_ASSIGNMENT_URL_TEMPLATE.buildWithQuery(api.getBaseURL(), builder.toString());
+        BoxAPIRequest request = new BoxAPIRequest(api, url, HttpMethod.GET);
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+
+        BoxStoragePolicyAssignment storagePolicyAssignment = new BoxStoragePolicyAssignment(api,
+                response.getJsonObject().get("entries").asArray().get(0).asObject().get("id").asString());
+        BoxStoragePolicyAssignment.Info info = storagePolicyAssignment.new
```

Comment:
This pattern is very yucky — we should try to eliminate this weirdness in the future